source/iterator/snapshot.go

```diff
@@ -0,0 +1,246 @@
+// Copyright © 2022 Meroxa, Inc & Yalantis.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package iterator
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	"github.com/huandu/go-sqlbuilder"
+	"github.com/jmoiron/sqlx"
+
+	"github.com/conduitio-labs/conduit-connector-db2/coltypes"
+	"github.com/conduitio-labs/conduit-connector-db2/source/position"
+)
+
+// snapshotIterator - iterator which get snapshot data.
+// A "snapshot" is the state of a table data at a particular point in time when connector starts work.
+// First time when the snapshot iterator starts work, it is get max value from `orderingColumn` and saves
+// this value to position.
+// The snapshot iterator reads all rows, where `orderingColumn` values less or equal maxValue,
+// from the table in batches.
+// Values in the ordering column must be unique and suitable for sorting, otherwise, the snapshot won't work correctly.
+// Iterators saves last processed value from `orderingColumn` column to position to field `SnapshotLastProcessedVal`.
+// If snapshot stops it will parse position from last record and will
+// try gets row where `{{orderingColumn}} > {{position.SnapshotLastProcessedVal}}`.
+type snapshotIterator struct {
+	db   *sqlx.DB
+	rows *sqlx.Rows
+
+	// table - table name.
+	table string
+	// columns list of table columns for record payload
+	// if empty - will get all columns.
+	columns []string
+	// keys Names of columns what iterator use for setting key in record.
+	keys []string
+	// orderingColumn Name of column what iterator using for sorting data.
+	orderingColumn string
+	// maxValue from ordering column.
+	maxValue any
```

Comment:
It's worth mentioning for what do we use this value.

Reply:
Yes , added